sources/web/datalab/static/datalab.js

```diff
@@ -80,15 +80,29 @@ function initializeDataLab(
     });
   }
   else if (pageClass.indexOf('notebook_list') >= 0) {
+    // The page is finished loading after the notebook list is drawn for the first
+    // time. The list is refreshed periodically though, so we need to only capture
+    // the first occurrence
+    let notebookListLoaded = false;
     events.on('draw_notebook_list.NotebookList', function() {
-      notebooklist.postLoad(ipy, ipy.notebook_list, ipy.new_notebook_widget,
-                             events, dialog, utils);
-      window.datalab.loaded = true;
+      if (!notebookListLoaded) {
+        notebooklist.postLoad(ipy, ipy.notebook_list, ipy.new_notebook_widget,
+                              events, dialog, utils);
+        window.datalab.loaded = true;
```

Comment:
Since you are already setting window.datalab.loaded, is there a reason to also set notebookListLoaded? Will there ever be a time when one is set but not the other?
Likewise for sessionListLoaded below.